### PR TITLE
Broken Promise response header propagation

### PR DESCRIPTION
**Pull request by cmark**

### CODE REVIEW

File: core/com.b2international.snowowl.core.rest/src/com/b2international/snowowl/core/rest/util/PromiseMethodReturnValueHandler.java [hunk 1]
```diff
@@ -51,23 +54,10 @@ public void handleReturnValue(Object returnValue, MethodParameter returnType, Mo
...
-						}
-						result.setResult(response);
-					}
-					return null;
-				})
+				.thenRespond(promiseResponse -> setDeferredResult(result, promiseResponse))
 				.fail(err -> {
 					if (result.isSetOrExpired()) {
-						LOG.warn("Deferred result is already set or expired, could not deliver Throwable.", err);
+						LOG.warn("Deferred result is already set or expired, could not deliver Throwable {}.", err);
```
Comment (apeteri) [inline core/com.b2international.snowowl.core.rest/src/com/b2international/snowowl/core/rest/util/PromiseMethodReturnValueHandler.java]: Without the `{}` placeholder in the format string we get the `Throwable`'s message and a stack trace, with it included the output will consist of whatever `Throwable#toString` or its overridden implementation would like to tell us (typically the class name and message, or just the message).

**Review by apeteri — APPROVED** [review]
👍 with a small note.

File: core/com.b2international.snowowl.core.rest/src/com/b2international/snowowl/core/rest/util/PromiseMethodReturnValueHandler.java (hunk 1 above)
Comment (cmark) [reply core/com.b2international.snowowl.core.rest/src/com/b2international/snowowl/core/rest/util/PromiseMethodReturnValueHandler.java]: Right, removing it.

Comment (cmark) [reply core/com.b2international.snowowl.core.rest/src/com/b2international/snowowl/core/rest/util/PromiseMethodReturnValueHandler.java]: Done

**Review by nagyo — APPROVED** [review]
LGTM!

**Pull request merged**